### PR TITLE
Add macros to instantiate children and supervisors

### PR DESCRIPTION
**Pull request by Stupremee**
#### Checklist

- [x] tests are passing with `cargo test`. 
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message is clear

This addition was requested in #3

### CODE REVIEW

File: bastion/src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    (@sort, $strat:expr, $cbs:expr,) => {
+        $crate::Bastion::supervisor(|sp| {
+            sp
+                .with_callbacks($cbs)
+                .with_strategy($strat)
+        }).expect("failed to create supervisor");
+    };
+}
+
+/// Spawnes a blocking task, which will be run on an extra thread pool, and
```
Comment (vertexclique) [inline bastion/src/macros.rs]: A couple of typos there, let's rephrase:
```
Spawns a blocking task, which will run on the blocking thread pool, and returns the handle.
```

File: bastion/src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                        }
+                    }
+                })
+        }).expect("failed to create children group");
+    };
+}
+
+/// This macro creates a new supervisor with the given strategy and the given callbacks.
+/// Children can be specified by using the `children` / `child` macro.
+/// You can provide as many children groups as you want. Supervised supervisors are currently not
```
Comment (vertexclique) [inline bastion/src/macros.rs]: ```Supervised supervisors are currently not yet supported.``` :)

File: bastion/src/macros.rs [hunk 3]
```diff
@@ -0,0 +1,206 @@
+//! This module contains some useful macros to simply
+//! create supervisors and children groups.
+
+/// This macro creates a new children group with the given amount of workers
+/// callbacks, and a closure which should be executed when a message is received.
```
Comment (vertexclique) [inline bastion/src/macros.rs]:
```
This macro creates a new children group with the given amount of worker
callbacks, and a closure that will be executed when a message is received.
```

File: bastion/src/macros.rs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+/// # Example
+/// ```
+/// # use std::{thread, time};
+/// # use lightproc::proc_stack::ProcStack;
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let task = blocking! {
+///     thread::sleep(time::Duration::from_millis(3000));
+/// };
+/// bastion_executor::run::run(task, ProcStack::default());
```
Comment (vertexclique) [inline bastion/src/macros.rs]: So the `run` method is basically `block_on` for any other systems around.
If possible let's create `run` macro too, I forgot that.

File: bastion/src/macros.rs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+/// let future2 = async move {
+///     10 / 2
+/// };
+///
+/// let result = run!(future2);
+/// assert_eq!(result, 5);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! run {
```
Comment (azastrael) [inline bastion/src/macros.rs]: Could you also add a `spawn!` macro that would call `bastion_executor::pool::spawn`, please? :)

**Pull request merged**